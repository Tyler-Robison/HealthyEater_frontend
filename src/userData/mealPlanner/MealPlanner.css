.MealPlanner {
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: rgb(155, 193, 204);
    flex-grow: 1;
}

.MealPlanner-header {
    margin: 1rem 0px;
    background-color: white;
    border-radius: 1rem;
    box-shadow: 2px 2px 8px rgba(0, 0, 0, 1);
    font-size: 1rem;
}

.MealPlanner-explain-para {
    margin: 0.4rem 0;
}

/* .MealPlanner-week-div { */
    /* width: 100vw; */
    /* height: 40vh; */

    /* border-top: 1px solid black;
    border-bottom: 1px solid black; */
    /* margin: auto; */
    /* margin-bottom: 1rem; */
/* } */

@media (max-width: 1000px) {
    .week-container {
        max-width: 500px !important;
    }
}

@media (max-width: 468px) {
    .MealPlanner-explain-para {
        font-size: 12px;
    }
}

.MealPlanner-form {
    padding-top: 10px;
    padding-bottom: 20px;
    height: 120px;
}

@media (max-width: 800px) {
    .MealPlanner-form {
        font-size: 12px;
    }
}

.no-recipe-para {
    font-size: 24px;
    margin-top: 3rem;
}

