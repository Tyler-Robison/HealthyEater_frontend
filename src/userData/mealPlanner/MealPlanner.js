import './MealPlanner.css'
import React, { useContext } from 'react'
import Day from './Day'
import PlanForm from './PlanForm'
import GlobalContext from '../../context/GlobalContext'
import PlannerAPI from '../../APIs/plannerAPI'
import { Link } from 'react-router-dom'

/** Mealplanner allows users to set their saved recipes into days of the week
 * 
 * Calculates daily and weekly points used based on recipes inputted into planner
 * 
 * renders PlanForm, the component used to add meals to mealplanner
 * 
 * Individual meals can be deleted by clicking on the red 'X' button, all meals can be deleted with Clear Calendar button*/
const MealPlanner = () => {
    const { currentUser, setCurrentUser, token } = useContext(GlobalContext);
    const days = ['Mon', 'Tues', 'Wed', 'Thurs', 'Fri', 'Sat', 'Sun'];

    const calculateWeeksPoints = () => currentUser.mealplan.reduce((accum, nextEle) => {
            return accum += nextEle.ww_points
        }, 0)
    

    const deleteMealplan = async () => {
        await PlannerAPI.deleteAllMeals(currentUser.id, token);
        currentUser.mealplan = [];
        setCurrentUser({ ...currentUser, currentUser });
    }

    if (currentUser.recipes.length === 0) return <p className="no-recipe-para">You have no saved recipes, go to <Link to={`/findrecipes`} className='RecipeList-para'><b>Find Recipes</b></Link></p>

    return (
        <div className='MealPlanner'>
            <div className='row d-flex flex-column align-items-center mt-3'>
                {/* if user hasn't calculated their points they can still use mealplanner but points functionality won't be there */}
                {!currentUser.points ? <p className='no-recipe-para'>Use <Link to={`/points`}><b>Point Calculator</b></Link> to determine points allotment </p> :
                    <div className='MealPlanner-header col-12'>
                        {currentUser.points && <h3>Weekly points used: {calculateWeeksPoints()}</h3>}
                        {currentUser.points && <p className='MealPlanner-explain-para'>Point Allowance - Daily: {currentUser.points}, Weekly: {currentUser.points * 7} Bonus: {Math.floor(currentUser.points * 1.4)}</p>}
                        {currentUser.points && <p className='MealPlanner-explain-para'>Bonus points can be used during the week for days when you exceed {currentUser.points} points</p>}
                        {currentUser.points && <p className='MealPlanner-explain-para'>Bonus points don't count towards weekly total</p>}
                    </div>}
            </div>

            <button className='general-btn-red mb-2' onClick={deleteMealplan}>Clear Calendar</button>

            <div className=' week-container container vw-100 mx-2'>
                <div className="MealPlanner-week-div row">
                    {days.map(day => <Day key={day} day={day} />)}
                </div>
            </div>

            <PlanForm days={days} />

        </div>
    )
}

export default MealPlanner;