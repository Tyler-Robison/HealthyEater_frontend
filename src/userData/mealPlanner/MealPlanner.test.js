import { render, screen } from '@testing-library/react';
import MealPlanner from './MealPlanner';
import { MemoryRouter } from "react-router-dom";
import ContextProvider from '../../testContext';

const days = ['Mon', 'Tues', 'Wed', 'Thurs', 'Fri', 'Sat', 'Sun'];

it("renders without crashing", function () {
   <MemoryRouter>
      <MealPlanner days={days} />
   </MemoryRouter>
});


it("matches snapshot, correct displaying 2 meals Monday and 1 Tuesday", function () {

   const container = render(
      <MemoryRouter>
         <ContextProvider >
            <MealPlanner days={days} />
         </ContextProvider>
      </MemoryRouter>
   );
   expect(container.asFragment()).toMatchSnapshot();
});




