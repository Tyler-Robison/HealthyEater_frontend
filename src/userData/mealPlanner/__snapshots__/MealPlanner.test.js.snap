// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, correct displaying 2 meals Monday and 1 Tuesday 1`] = `
<DocumentFragment>
  <div
    class="MealPlanner"
  >
    <div
      class="row d-flex flex-column align-items-center mt-3"
    >
      <p
        class="no-recipe-para"
      >
        Use 
        <a
          href="/points"
        >
          <b>
            Point Calculator
          </b>
        </a>
         to determine points allotment 
      </p>
    </div>
    <button
      class="general-btn-red mb-2"
    >
      Clear Calendar
    </button>
    <div
      class=" week-container container vw-100 mx-2"
    >
      <div
        class="MealPlanner-week-div row"
      >
        <div
          class="Day col-lg d-flex flex-column"
        >
          <p>
            <b>
              Mon
            </b>
          </p>
          <div
            class="list-container flex-grow-1"
          >
            <ol>
              <li>
                 
                <a
                  class="RecipeList-para"
                  href="/recipes/1111"
                >
                  <b>
                    bacon
                  </b>
                </a>
                 (20 points)
                <button
                  class="general-btn-red small-btn"
                  data-testid="1"
                >
                  X
                </button>
              </li>
              <li>
                 
                <a
                  class="RecipeList-para"
                  href="/recipes/2222"
                >
                  <b>
                    eggs
                  </b>
                </a>
                 (10 points)
                <button
                  class="general-btn-red small-btn"
                  data-testid="2"
                >
                  X
                </button>
              </li>
            </ol>
          </div>
          <p
            class="Day-points"
          >
            <b>
              Points:
            </b>
             30
          </p>
        </div>
        <div
          class="Day col-lg d-flex flex-column"
        >
          <p>
            <b>
              Tues
            </b>
          </p>
          <div
            class="list-container flex-grow-1"
          >
            <ol>
              <li>
                 
                <a
                  class="RecipeList-para"
                  href="/recipes/3333"
                >
                  <b>
                    salad
                  </b>
                </a>
                 (5 points)
                <button
                  class="general-btn-red small-btn"
                  data-testid="3"
                >
                  X
                </button>
              </li>
            </ol>
          </div>
          <p
            class="Day-points"
          >
            <b>
              Points:
            </b>
             5
          </p>
        </div>
        <div
          class="Day col-lg d-flex flex-column"
        >
          <p>
            <b>
              Wed
            </b>
          </p>
          <div
            class="list-container flex-grow-1"
          >
            <ol />
          </div>
        </div>
        <div
          class="Day col-lg d-flex flex-column"
        >
          <p>
            <b>
              Thurs
            </b>
          </p>
          <div
            class="list-container flex-grow-1"
          >
            <ol />
          </div>
        </div>
        <div
          class="Day col-lg d-flex flex-column"
        >
          <p>
            <b>
              Fri
            </b>
          </p>
          <div
            class="list-container flex-grow-1"
          >
            <ol />
          </div>
        </div>
        <div
          class="Day col-lg d-flex flex-column"
        >
          <p>
            <b>
              Sat
            </b>
          </p>
          <div
            class="list-container flex-grow-1"
          >
            <ol />
          </div>
        </div>
        <div
          class="Day col-lg d-flex flex-column"
        >
          <p>
            <b>
              Sun
            </b>
          </p>
          <div
            class="list-container flex-grow-1"
          >
            <ol />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MealPlanner-form"
    >
      <form>
        <label
          for="daySelect"
        >
          Select Day
        </label>
        <select
          id="daySelect"
          name="daySelect"
        >
          <option
            value="Mon"
          >
            Mon
          </option>
          <option
            value="Tues"
          >
            Tues
          </option>
          <option
            value="Wed"
          >
            Wed
          </option>
          <option
            value="Thurs"
          >
            Thurs
          </option>
          <option
            value="Fri"
          >
            Fri
          </option>
          <option
            value="Sat"
          >
            Sat
          </option>
          <option
            value="Sun"
          >
            Sun
          </option>
        </select>
        <label
          for="mealSelect"
        >
          Select Meal
        </label>
        <select
          id="mealSelect"
          name="mealSelect"
        >
          <option
            value="{\\"name\\":\\"bacon\\",\\"recipe_id\\":1111,\\"ww_points\\":20}"
          >
            bacon
          </option>
          <option
            value="{\\"name\\":\\"eggs\\",\\"recipe_id\\":2222,\\"ww_points\\":10}"
          >
            eggs
          </option>
          <option
            value="{\\"name\\":\\"salad\\",\\"recipe_id\\":3333,\\"ww_points\\":5}"
          >
            salad
          </option>
          <option
            value="{\\"name\\":\\"bread\\",\\"recipe_id\\":4444,\\"ww_points\\":7}"
          >
            bread
          </option>
        </select>
        <button
          class="general-btn"
          type="submit"
        >
          Add to Calendar
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
